GestureHandler.js

```diff
@@ -153,6 +153,63 @@ function filterConfig(props, validProps, defaults = {}) {
   });
   return res;
 }
+function backwardCompatibleTransformProps(handlerName, props) {
+  const res = {};
+  if (handlerName === 'PanGestureHandler') {
```

Comment:
I'd recommend that we add additional callback to `createHandler` called `transformProps` and then call it in `componentDidMount`:
```
this._config = filterConfig(
      transformProps(this.props),
      this.constructor.propTypes,
      config
);
```

Then for `PanHandler` we can have a transform that translates old props into new style props and for other handlers that callback would be empty (no props transforming).

Another thing is that I'd like to have that code which checks for props compatibility wrapped in `__DEV__` so that it only runs in dev mode and use `console.warning` to report.

Finally, I'd like if we could add "shortcut" props for defining the range and allow `minOffsetRange` to be an array `[start, end]`